atlasdb-commons/src/main/java/com/palantir/util/debug/StackTraceUtils.java

```diff
@@ -258,11 +300,12 @@ private static int score(String trace) {
             int nextStartingIndex = index + 1;
             index = trace.indexOf("com.palantir", nextStartingIndex);
         }
-        score += POINTS_PER_LINE * trace.split(LINE_ENDING).length;
+        score += POINTS_PER_LINE * trace.chars().filter(x -> x == '\n').sum();
```

Comment:
Hmm, this isn't quite right I think:
- it should be `count()`, not `sum()` - which I think adds the character values (so that gives us an answer that is 10X larger than it should be because `\n` is `10`).
- the semantics are different in the presence of trailing newlines; the original doesn't count them.

```
        String s = "a\nb\nc\nd\n";
        System.out.println(s.split(LINE_ENDING).length); // 4
        System.out.println(s.chars().filter(c -> c == '\n').sum()); // 40
        System.out.println(s.chars().filter(c -> c == '\n').count()); // 4

        s = "\na\n\nfoo\n\n\nbar";
        System.out.println(s.split(LINE_ENDING).length); // 7
        System.out.println(s.chars().filter(c -> c == '\n').count()); // 6

        s = "\na\n\nfoo\n\n\nbar\n\n\n\n\n\n\n\n";
        System.out.println(s.split(LINE_ENDING).length); // 7
        System.out.println(s.chars().filter(c -> c == '\n').count()); // 14

        s = "\n\n\n\na\n\nfoo\n\n\nbar";
        System.out.println(s.split(LINE_ENDING).length); // 10
        System.out.println(s.chars().filter(c -> c == '\n').count()); // 9
```

To get a perfect reproduction of past behaviour I think we need to strip all trailing newlines, then count the remaining newlines and add 1. I'm not sure how precisely we need the behaviour to stay the same, but given the scary warnings at the top I would be careful.

Reply:
Oops, this should definitely be `count()`.

In terms of the differences between split & counting new lines, I'm not sure it matters too much in the case of scoring. I can update with a hand rolled method & tests to count non-blank lines without allocating when I get cycles.

```
static int countNonBlankLines(String value) {
    if (value == null || value.isEmpty()) {
        return 0;
    }

    int count = 0;
    int index = 0;
    int previousIndex = 0;
    while ((index = value.indexOf('\n', previousIndex) > -1) {
        if (index > previousIndex + 1) {
            count++;
        }
        previousIndex = index;
    }
    return count;
}
```